rero_ils/modules/acq_orders/serializers/csv.py

```diff
@@ -0,0 +1,283 @@
+from rero_ils.modules.documents.api import DocumentsSearch
+from rero_ils.modules.documents.serializers.base import DocumentFormatter
+from rero_ils.modules.vendors.api import VendorsSearch
+from rero_ils.utils import get_i18n_supported_languages
+
+creator_role_filter = [
+    'rsp', 'cre', 'enj', 'dgs', 'prg', 'dsr', 'ctg', 'cmp', 'inv', 'com',
+    'pht', 'ivr', 'art', 'ive', 'chr', 'aut', 'arc', 'fmk', 'pra', 'csl'
+]
+
+
+class AcqOrderCSVSerializer(CSVSerializer):
+    """Mixin serializing records as CSV."""
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        :param item_links_factory: Factory function for record links.
+        """
+        # define chunk size
+        chunk_size = 100
+        vendors = {}
+
+        # language
+        language = request.args.get("lang", current_i18n.language)
+        if not language or language not in get_i18n_supported_languages():
+            language = current_app.config.get('BABEL_DEFAULT_LANGUAGE', 'en')
+
+        order_fields = {
+            'order_pid': 'pid',
+            'order_reference': 'reference',
+            'order_date': 'order_date',
+            'order_type': 'type',
+            'order_status': 'status'
+        }
+
+        def generate_csv():
+            def batch(results):
+                """Chunk search results.
+
+                :param results: search results.
+                :return list of chunked item pids and search records
+                """
+                records = {}
+                pids = []
+                for result in results['hits']['hits']:
+                    data = result['_source']
+                    pids.append(data['pid'])
+                    records[data['pid']] = data
+                    if len(records) % chunk_size == 0:
+                        yield pids, records
+                        pids = []
+                        records = {}
+                yield pids, records
+
+            def get_linked_records_by_order_pids(order_pids):
+                """Get linked resources for the given order pids."""
+                order_line_fields = [
+                    'pid', 'quantity', 'amount', 'total_amount', 'status',
+                    'priority', 'notes', 'document', 'acq_account',
+                    'acq_order'
+                ]
+                records = AcqOrderLinesSearch() \
+                    .get_records_by_terms(terms=order_pids,
+                                          key='acq_order__pid',
+                                          fields=order_line_fields)
+                order_lines = {}
+                doc_pids = []
+                account_pids = []
```

Comment:
same here

Reply:
`order_lines, doc_pids, account_pid = {}, [], []` is less readable for me but OK...